ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/keystoneqr/viewmodel/KeystoneQrViewModel.kt

```diff
@@ -0,0 +1,57 @@
+package co.electriccoin.zcash.ui.screen.keystoneqr.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import cash.z.ecc.sdk.ANDROID_STATE_FLOW_TIMEOUT
+import co.electriccoin.zcash.ui.screen.keystoneqr.state.KeystoneQrState
+import com.keystone.module.SolSignRequest
+import com.keystone.sdk.KeystoneSDK
+import com.keystone.sdk.KeystoneSolanaSDK
+import com.sparrowwallet.hummingbird.UREncoder
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.WhileSubscribed
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+
+class KeystoneQrViewModel : ViewModel() {
+    private val qr = genSolanaQRCode()
+
+    private val currentQrPart = MutableStateFlow(qr.nextPart())
+
+    val state =
+        currentQrPart.map {
+            KeystoneQrState(
+                qrData = it,
+                generateNextQrCode = {
+                    currentQrPart.update { qr.nextPart() }
+                }
+            )
+        }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(ANDROID_STATE_FLOW_TIMEOUT), null)
+
+    @Suppress("MaxLineLength", "MagicNumber")
+    private fun genSolanaQRCode(): UREncoder {
```

Comment:
Let's move this sample code to a fixture instead of the production part so we can easily replace it later.

Reply:
this is refactored in the next coming PR